File: bubblesort.c
#include<stdio.h>
void main()
{
	int i,j,temp,n;
	printf("Enter number of elements : ");
	scanf("%d",&n);
	int ar[n];
	printf("\nEnter %d elements \n",n);
	for(i=0;i<n;i++)
	{
		printf("Enter element %d : ",i+1);
		scanf("%d",&ar[i]);
	}
	for(i=0;i<n;i++)
	{
		for(j=0;j<n-i-1;j++)
		{
			if(ar[j]>ar[j+1])
			{
				temp=ar[j];
				ar[j]=ar[j+1];
				ar[j+1]=temp;
			}
		}
	}
	printf("\n\nThe sorted array is : \n");
	for(i=0;i<n;i++)
		printf("Element %d : %d\n",i+1,ar[i]);
}

File: python_DSA problems/bubblesort.py
def bubble_sort(arr):
    n = len(arr)
    
    for i in range(n):
        # Last i elements are already in place, no need to check them
        for j in range(0, n-i-1):
            # Swap if the element found is greater than the next element
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

# Take user input for the array
user_input = input("Enter the elements of the array (separated by spaces): ")
user_array = list(map(int, user_input.split()))

# Call the bubble sort function
bubble_sort(user_array)

# Print the sorted array
print("Sorted array:", user_array)
